TestSuite/utils.py

```diff
@@ -1,55 +1,28 @@
 class IsEqualFunctions:
     @staticmethod
-    def is_dicts_equal(dict1: dict, dict2: dict):
-        if not isinstance(dict1, dict) or not isinstance(dict2, dict):
-            return False
-
-        if sorted(list(dict1.keys())) != sorted(list(dict2.keys())):
-            return False
-
-        for key, dict1_value in dict1.items():
-            dict2_value = dict2.get(key)
-
-            if not isinstance(dict2_value, type(dict1_value)):
-                return False
+    def is_dicts_equal(dict1: dict, dict2: dict, simple_comparison: bool):
+        dict1 = dict(dict1)
+        dict2 = dict(dict2)
 
-            if isinstance(dict1_value, dict):
-                if not IsEqualFunctions.is_dicts_equal(dict1_value, dict2_value):
-                    return False
+        if simple_comparison:
```

Comment:
What does it mean? Comparing the pointers?
Is it different than below where all values are not `dict`/`list`/`set`?
If it's the same, then there's a potential for extra work...

Reply:
dict1==dict2 might never work. it'll work only if it is the same instance (like `dict is dict`)

Reply:
for dict diff, we can use this tool:
https://github.com/seperman/deepdiff

it has only subdependency (ordered-dict) which is fine to add to the requirements.

Reply:
also, no need to use it as Pytest magic works on dict equality: https://stackoverflow.com/questions/47201358/how-can-you-test-that-two-dictionaries-are-equal-with-pytest-in-python

Reply:
@ilaner and I checked the variety of functions using this one, and found out the important part is the ability to compare lists as sets in specific cases (mostly when comparing a list that is created from a set), so we updated the function. Please check out the new version, it's pretty straightforward and doesn't require external tools.